java/code/src/com/suse/manager/webui/services/FutureUtils.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2020 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.manager.webui.services;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Class providing utility methods for futures.
+ */
+public class FutureUtils {
+
+    private static final ScheduledExecutorService SCHEDULED_EXECUTOR_SERVICE =
+            Executors.newScheduledThreadPool(1);
```

Comment:
What motivated the choice of using one thread here, given original code allocated 5?

https://github.com/uyuni-project/uyuni/blob/4.1-alpha1/java/code/src/com/suse/manager/webui/services/impl/SaltService.java#L155

Unrelated note: `SaltService.scheduledExecutorService` seems to be dead code now.

Reply:
reasoning was that this thread pool is not used for heavy long running computation but only to set a future to failed so 1 thread should be enough for that.